src/complex/DataStructure/DataObject.cpp

```diff
@@ -169,16 +169,34 @@ bool DataObject::canRename(const std::string& name) const
     return false;
   }
 
-  auto dataStruct = getDataStructure();
-  if(dataStruct == nullptr)
+  if(name == getName())
+  {
+    return true;
+  }
+
+  const auto* dataStructPtr = getDataStructure();
+  if(dataStructPtr == nullptr)
   {
     return false;
   }
-  return !std::any_of(m_ParentList.cbegin(), m_ParentList.cend(), [dataStruct, name](IdType parentId) { return dataStruct->getDataAs<BaseGroup>(parentId)->contains(name); });
+
+  return !std::any_of(m_ParentList.cbegin(), m_ParentList.cend(), [dataStructPtr, name](IdType parentId) {
+    const auto* baseGroupPtr = dataStructPtr->getDataAs<BaseGroup>(parentId);
+    if(baseGroupPtr == nullptr)
+    {
+      std::cout << "DataObject::canRename(name=" << name << ") cannot get baseGroup from parentId = " << parentId << std::endl;
+    }
+    return baseGroupPtr != nullptr && baseGroupPtr->contains(name);
+  });
 }
 
 bool DataObject::rename(const std::string& name)
 {
+  if(name == m_Name)
+  {
+    return true;
+  }
+
```

Comment:
```suggestion
```

This check feels a bit redundant since we check the exact same thing in the canRename function which gets called immediately after.

Reply:
I sort of agree. Let me take another look at it.